scripts/mutect2_wdl/unsupported/mutect2_opt.wdl

```diff
@@ -72,25 +88,30 @@ workflow Mutect2 {
       docker = basic_bash_docker
   }
 
+  Int split_interval_size = ref_size + ceil(size(intervals, "GB") * small_input_to_output_mulitplier) + disk_pad
 
   call SplitIntervals {
     input:
-      gatk4_jar = gatk4_jar,
       scatter_count = scatter_count,
       intervals = intervals,
       ref_fasta = ref_fasta,
       ref_fasta_index = ref_fasta_index,
       ref_dict = ref_dict,
       gatk4_jar_override = gatk4_jar_override,
       preemptible_attempts = preemptible_attempts,
-      m2_docker = m2_docker
-
+      m2_docker = m2_docker,
+      disk_space_gb = split_interval_size
   }
 
+  # Size M2 differently based on if we are using NIO or not
+  Int m2_output_size = tumor_bam_size / scatter_count
+  Int m2_nio_disk_size = (tumor_bam_size + normal_bam_size) / scatter_count + ref_size + gnomad_vcf_size + m2_output_size + disk_pad
+  Int m2_no_nio_disk_size = tumor_bam_size + normal_bam_size + ref_size + gnomad_vcf_size + m2_output_size + disk_pad
+  Int m2_per_scatter_size = if defined(auth) then m2_nio_disk_size else m2_no_nio_disk_size
```

Comment:
Nice, this is super clear. 

If gnomad is also using NIO can you divide `gnomad_vcf_size` by `scatter_count` as well?

Also, can you just divide by `scatter_count` instead of something smaller than `scatter_count`? Or does the `disk_pad` take care of any unevenness in the shard sizes?

Reply:
oh right! yea i made disk pad larger than we did in single sample and I'm over estimating `m2_output_size` to take care of generating and not generating a bamout.  Lots of oversizing here i guess =/